src/tt_store.erl

```diff
@@ -1,9 +1,94 @@
 -module(tt_store).
 
--export([find_by_time/2]).
+-behaviour(gen_server).
+-define(SERVER, ?MODULE).
 
--record(talk, {title :: string(), start_time :: calendar:datetime(), end_time :: calendar:datetime(), location :: string()}).
+-include_lib("stdlib/include/ms_transform.hrl").
+%% API
+-export([start_link/0, add/4, find_by_time/2, list/0, find_by_name/1]).
 
--spec(find_by_time(calendar:datetime(), calendar:datetime()) -> [#talk{}]).
-find_by_time(_StartTime, _EndTime) ->
-  [#talk{title = "Title", start_time = calendar:local_time(), end_time = calendar:local_time(), location =  "Location"}].
+%% Supervisor callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+%% DB record
+-record(talk, {title :: string(), startTime :: {Hour, Minute, Second}, endTime :: {Hour, Minute, Second}, location :: any()}).
+
+%%====================================================================
+%% API functions
+%%====================================================================
+
+start_link() ->     
+    gen_server:start_link({local, ?SERVER},?MODULE, [], []).
+
+add(Title, StartTime, EndTime, Where) -> 
+    gen_server:cast(?SERVER, {add, {Title, StartTime, EndTime, Where}}),
+    ok.
+
+find_by_time(StartTime, EndTime) -> 
+    gen_server:call(?SERVER, {find_time, {StartTime, EndTime}}).
+
+list() -> 
+    gen_server:cast(?SERVER, list).
+
+find_by_name(Name) ->
+    gen_server:call(?SERVER, {find_name, Name}).
+%%====================================================================
+%% Supervisor callbacks
+%%====================================================================
+init([]) -> 
+    ets:new(talks, [set, named_table]),
+    process_flag(trap_exit, true),
+    {ok, talks}.
+
```

Comment:
You shouldn't be putting the `talks` atom directly as the state in the return value of `handle_call/3` or `handle_cast/2`. If the state doesn't change just pass it over. So better implementation would be:

```erlang
 Comment Cancel
 handle_call({find_time, {StartTime, EndTime}}, _From, State) -> 
     Records = find_records_by_time(State, {StartTime, EndTime}),    
     {reply, Records, State};
```

Alternatively, treat the name of the table as a constant and put it in the macro:

```erlang
-define(TABLE, talks).
```

Then refer to the table like this:

```erlang
Records = ets:match(?TABLE, '$1')
```